config/ompi_setup_java.m4

```diff
@@ -13,92 +13,238 @@ dnl                         All rights reserved.
+                      # too.  Ugh.
+                      AS_IF([test -d "$with_jdk_headers/darwin"],
+                            [OMPI_JDK_CPPFLAGS="$OMPI_JDK_CPPFLAGS -I$with_jdk_headers/darwin"])
+
+                      CPPFLAGS="$CPPFLAGS $OMPI_JDK_CPPFLAGS"])
+               AC_CHECK_HEADER([jni.h], [],
+                               [ompi_java_happy=no])
+               CPPFLAGS=$ompi_java_CPPFLAGS_save
+              ])
     else
-        AC_MSG_RESULT([no])
-        WANT_MPI_JAVA_SUPPORT=0
+        ompi_java_happy=no
     fi
-    AC_DEFINE_UNQUOTED([OMPI_WANT_JAVA_BINDINGS], [$WANT_MPI_JAVA_SUPPORT],
-                       [do we want java mpi bindings])
-    AM_CONDITIONAL(OMPI_WANT_JAVA_BINDINGS, test "$WANT_MPI_JAVA_SUPPORT" = "1")
-
-   # Are we happy?
-    AS_IF([test "$WANT_MPI_JAVA_SUPPORT" = "1"],
-          [AC_MSG_WARN([******************************************************])
-           AC_MSG_WARN([*** Java MPI bindings are provided on a provisional])
-           AC_MSG_WARN([*** basis.  They are NOT part of the current or])
-           AC_MSG_WARN([*** proposed MPI standard.  Continued inclusion of])
-           AC_MSG_WARN([*** the Java MPI bindings in Open MPI is contingent])
-           AC_MSG_WARN([*** upon user interest and developer support.])
-           AC_MSG_WARN([******************************************************])
-          ])
-
-    AC_CONFIG_FILES([
-        ompi/mpi/java/Makefile
-        ompi/mpi/java/java/Makefile
-        ompi/mpi/java/c/Makefile
-    ])
+    AC_SUBST(OMPI_JDK_CPPFLAGS)
+
+    # Are we happy?
+    AC_MSG_CHECKING([if Java support available])
+    AS_IF([test "$ompi_java_happy" = "no"],
+          [AC_MSG_RESULT([no])],
+          [AC_MSG_RESULT([yes])])
+
+    OPAL_VAR_SCOPE_POP
+])
+
+dnl OMPI_SETUP_JAVA()
+dnl ----------------
+dnl Do everything required to setup the Java compiler.  Safe to AC_REQUIRE
+dnl this macro.
+AC_DEFUN([OMPI_SETUP_JAVA],[
+    OPAL_VAR_SCOPE_PUSH([ompi_java_happy ompi_javah_happy])
+
+    ompi_java_happy=no
+    ompi_javah_happy=no
+
+    AC_ARG_WITH(jdk-dir,
+                AC_HELP_STRING([--with-jdk-dir(=DIR)],
+                               [Location of the JDK header directory.  If you use this option, do not specify --with-jdk-bindir or --with-jdk-headers.]))
+    AC_ARG_WITH(jdk-bindir,
+                AC_HELP_STRING([--with-jdk-bindir(=DIR)],
+                               [Location of the JDK bin directory.  If you use this option, you must also use --with-jdk-headers (and you must NOT use --with-jdk-dir)]))
+    AC_ARG_WITH(jdk-headers,
+                AC_HELP_STRING([--with-jdk-headers(=DIR)],
+                               [Location of the JDK header directory.  If you use this option, you must also use --with-jdk-bindir (and you must NOT use --with-jdk-dir)]))
+
+    # Only setup the compiler if we were requested to
+    AS_IF([test "$1" = "yes"],
+          [_OMPI_SETUP_JAVA])
+
+    # JMS TO BE DELETED
```

Comment:
just to be clear, this only applies to `OMPI_HAVE_JAVA_SUPPORTBOGUS_TO_BE_DELETEDp`.
I also noticed the initial `OPAL_HAVE_HAVE_SUPPORT` _macro_ is unused, so as far as I am concerned, you can remove it now.

Reply:
Good catch; I meant to delete that before creating the PR.  Thanks.